### PR TITLE
bugfix: Always check if a Scala file is an actual file

### PR DESCRIPTION
**Pull request by tgodzik**
Fixes https://github.com/scalameta/metals/issues/4989

### CODE REVIEW

**Comment by kpodsiad** [comment]
I'm certain that I see this kind of fix recently. Is is another place?

**Review by ckipp01 — APPROVED** [review]
Looks simple enough. Do you think we should add a test for this though in MetalsTestEnrichments? However it might be way more involved to actually just test this. So if it's overkill, then just ignore this and LGTM

**Review by dos65 — APPROVED** [review]
:+1:

File: mtags/src/main/scala/scala/meta/internal/mtags/CommonMtagsEnrichments.scala [hunk 1]
```diff
@@ -518,12 +518,15 @@ trait CommonMtagsEnrichments {
      */
     def isScalaOrJava: Boolean = {
       toLanguage match {
-        case Language.SCALA | Language.JAVA =>
-          !Files.exists(path.toNIO) ||
-          Files.isRegularFile(path.toNIO)
+        case Language.SCALA | Language.JAVA => isFile
```
Comment (tgodzik) [inline mtags/src/main/scala/scala/meta/internal/mtags/CommonMtagsEnrichments.scala:521]: > I'm certain that I saw this kind of fix recently. Is is another place?

The fix was here and I didn't notice we had a single `isScala` and `isJava` also

**Comment by tgodzik** [comment]
> Looks simple enough. Do you think we should add a test for this though in MetalsTestEnrichments? However it might be way more involved to actually just test this. So if it's overkill, then just ignore this and LGTM

it's a rare scenario, so I am content just leaving it without tests for now

**Pull request merged**